### PR TITLE
Chapter 8: Basic Login

### PR DESCRIPTION
**Pull request by baaohuii**

### CODE REVIEW

**Comment by baaohuii** [comment]
ready

**Comment by baaohuii** [comment]
ready

File: .gitignore [hunk 1]
```diff
@@ -39,7 +39,7 @@
 /yarn-error.log
 yarn-debug.log*
 .yarn-integrity
-/config/database.yml
+#/config/database.yml
```
Comment (chungpv-1008) [inline .gitignore]: Why did not you ignore this file?

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -73,14 +73,14 @@ aside {
     &:first-child {
       border: 0;
       padding-top: 0;
-    }  
-
+    }    
+
```
Comment (chungpv-1008) [inline app/assets/stylesheets/custom.scss]:
```suggestion
    }

```

File: app/assets/stylesheets/users.scss [hunk 3]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Users controller here.
```
Comment (chungpv-1008) [inline app/assets/stylesheets/users.scss]: remove this file because not use it

File: app/controllers/users_controller.rb [hunk 4]
```diff
@@ -8,7 +8,7 @@ def new
   end
 
   def create
-    @user = User.new user_params # Not the final implementation!
+    @user = User.new(user_params) # Not the final implementation!
```
Comment (chungpv-1008) [inline app/controllers/users_controller.rb]:
```suggestion
    @user = User.new user_params
```

File: app/views/users/new.html.erb [hunk 5]
```diff
@@ -1,18 +1,19 @@
...
-            <%= pluralize @user.errors.count, "error" %>.
-          </div>
-          <ul>
-          <% @user.errors.full_messages.each do |msg| %>
-            <li><%= msg %></li>
-          <% end %>
+  <div class="row">
+    <div class="col-md-6 col-md-offset-3">
+      <%= form_for @user, url: signup_path do |f| %>
+
```
Comment (chungpv-1008) [inline app/views/users/new.html.erb]:
```suggestion
```

File: app/views/users/new.html.erb [hunk 6]
```diff
@@ -1,18 +1,19 @@
...
+        <% if @user.errors.any? %>
+          <div id="error_explanation">
+            <div class="alert alert-danger">
+              The form contains
+              <%= pluralize @user.errors.count, "error" %>
```
Comment (chungpv-1008) [inline app/views/users/new.html.erb]:
```suggestion
              <%= pluralize @user.errors.size, "error" %>
```

File: config/database.yml [hunk 7]
```diff
@@ -0,0 +1,16 @@
+default: &default
```
Comment (chungpv-1008) [inline config/database.yml]: ignore this file and create file 'database.yml.examble', then copy content from 'database.yml' to 'database.yml.examble'

File: config/routes.rb [hunk 8]
```diff
@@ -1,4 +1,5 @@
 Rails.application.routes.draw do
+  get 'users/show'
```
Comment (chungpv-1008) [inline config/routes.rb]: using " instead ', move this line to the bottom

File: app/assets/stylesheets/sessions.scss [hunk 9]
```diff
@@ -0,0 +1,3 @@
+// Place all the styles related to the Sessions controller here.
```
Comment (chungpv-1008) [inline app/assets/stylesheets/sessions.scss]: Delete this file because not use

File: app/controllers/sessions_controller.rb [hunk 10]
```diff
@@ -0,0 +1,20 @@
+class SessionsController < ApplicationController
+  def new
+  end
```
Comment (chungpv-1008) [inline app/controllers/sessions_controller.rb]:
```suggestion
  def new; end
```

File: app/helpers/sessions_helper.rb [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  def current_user
+    @current_user ||= User.find_by id: session[:user_id]
+  end
+
+  def logged_in?
+    current_user.present?
+  end
+
+  def log_out
+    session.delete(:user_id)
```
Comment (chungpv-1008) [inline app/helpers/sessions_helper.rb]:
```suggestion
    session.delete :user_id
```

File: app/views/sessions/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,14 @@
+<% provide(:title, "Log in") %>
```
Comment (chungpv-1008) [inline app/views/sessions/new.html.erb]: remove paratheses

File: app/views/sessions/new.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+<h1>Log in</h1>
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(:session, url: login_path) do |f| %>
```
Comment (chungpv-1008) [inline app/views/sessions/new.html.erb]: similar

File: app/views/sessions/new.html.erb [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+      <%= f.label :email %>
+      <%= f.email_field :email, class: 'form-control' %>
+      <%= f.label :password %>
+      <%= f.password_field :password, class: 'form-control' %>
```
Comment (chungpv-1008) [inline app/views/sessions/new.html.erb]: using " instead of '

File: config/database.yml [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+  socket: /var/run/mysqld/mysqld.sock    
+
+development:
+  <<: *default
+  database: sample_app_development
+
+test:
+  <<: *default
+  database: sample_app_test
+
```
Comment (chungpv-1008) [inline config/database.yml]: add endline for this file

**Comment by baaohuii** [comment]
ready

File: app/views/layouts/_header.html.erb [hunk 16]
```diff
@@ -1,12 +1,28 @@
...
+            <%= link_to "#", class: "dropdown-toggle", "data-toggle": "dropdown" do %>
+              <%= t("account") %>
+              <b class="caret"></b>
+            <% end %>
+            <ul class="dropdown-menu">
+	          <li><%= link_to t("profile"), current_user %></li>
+		      <li><%= link_to t("settings"), "#" %></li>
+		      <li class="divider"></li>
+		      <li><%= link_to t("logout"), logout_path, method: :delete %></li>
+	        </ul>
```
Comment (chungpv-1008) [inline app/views/layouts/_header.html.erb:20]: indent

File: app/views/users/new.html.erb [hunk 17]
```diff
@@ -1,18 +1,18 @@
...
-        <div id="error_explanation">
-          <div class="alert alert-danger">
-            <%= t "the_form_contains" %>
-            <%= pluralize @user.errors.count, "error" %>.
-          </div>
-          <ul>
-          <% @user.errors.full_messages.each do |msg| %>
-            <li><%= msg %></li>
-          <% end %>
+  <div class="row">
```
Comment (chungpv-1008) [inline app/views/users/new.html.erb:3]: indent

File: config/database.yml (hunk 7 above)
Comment (chungpv-1008) [reply config/database.yml]: you haven't ignored this file. Please add `config/database.yml` to file '.gitignore', then create the file 'database.yml.examble' and copy content from 'database.yml' to 'database.yml.examble'

**Review by chungpv-1008 — COMMENTED** [review]
Please fix comment, then merge the pull request yourself

**Comment by vanvtt-0952** [comment]
merged
Please fix comment, then merge the pull request yourself

**Pull request merged**